src/PendingCalls/TestCall.php

```diff
@@ -615,6 +622,20 @@ public function coversNothing(): self
         return $this;
     }
 
+    /**
+     * Adds a reference to the tested method or class. This helps to link test
+     * cases to the source code for easier navigation.
+     *
+     * @param  array<class-string|string>|class-string  ...$classes
+     */
+    public function references(string|array ...$classes): self
+    {
+        // For rector
+        $this->references = $classes;
```

Comment:
We could replace this with an `assert()`, this way we can remove the property for Rector (so that we aren't storing values for `$references` in memory).
```suggestion
        assert($classes !== []);
```

This would also be useful to have as we shouldn't really provide an empty call to `references()`, so the assertion would flag this up. 🤷🏻

Reply:
I think this is a better idea. it also reduces the load by itself, while adding an extra check to avoid using an empty array. thanks, Owen!